ISISComputingGroup/ibex_bluesky_core
# Add IBEX plotting callback

https://github.com/ISISComputingGroup/ibex_bluesky_core/issues/8

### Docs
- [ ] Ensure docs give enough information/pointers to documentation elsewhere for users to find out how to configure plots.
- [ ] Ensure all the points in the ticket's acceptance criteria have been met.

### To test this in a standalone session:
- Ensure you are in the `ibex_bluesky_core` venv as per dev setup notes.
- Ensure you have recently re-run `python -m pip install -e .[dev]` to pick up matplotlib and qt dependencies
- Run `python src\ibex_bluesky_core\demo_plan.py`
- [x] A matplotlib `Qt` window should appear and be updated live during the plan.
- [x] A long running plan like `bps.sleep(999)` should be interruptible via `ctrl-c`.

### To test this within the IBEX gui:
- Rebuild your `genie_python` using the branch linked from https://github.com/ISISComputingGroup/IBEX/issues/8453
- Run `c:\instrument\apps\python3\python.exe -m pip install -e c:\instrument\dev\ibex_bluesky_core\`
  * This installs `ibex_bluesky_core` into your genie_python environment (not a `.venv`)
- Ensure IBEX is set up with blocks and DAE as per existing dev setup notes.
- In the IBEX pydev scripting console run:
```python
from ibex_bluesky_core.run_engine import get_run_engine
from ibex_bluesky_core.demo_plan import demo_plan
RE = get_run_engine()
RE(demo_plan())
```
- [x] A matplotlib window should appear and be updated live during the plan.
- [ ] A long running plan like `bps.sleep(999)` should be interruptible via `ctrl-c`.
